src/interactable_music_plant_keyboard.c

```diff
@@ -0,0 +1,323 @@
+            } else if (kb->unkE < 0) {
+                if (LEVEL_TO_ZONE(gCurrentLevel) == ZONE_6) {
+                    gPlayer.speedAirX = -sKeyboardAccelTechnoBase[1][0];
+                    gPlayer.speedAirY = -sKeyboardAccelTechnoBase[1][1];
+                } else {
+                    gPlayer.speedAirX = -sKeyboardAccelMusicPlant[1][0];
+                    gPlayer.speedAirY = -sKeyboardAccelMusicPlant[1][1];
+                }
+
+                r7 = Q_8_8(7. / 8.);
+                r5 = -Q_8_8(4.5);
+                r8 = -Q_8_8(7. / 8.);
+                r6 = r5;
+            } else {
+                if (LEVEL_TO_ZONE(gCurrentLevel) == ZONE_6) {
+                    gPlayer.speedAirX = -sKeyboardAccelTechnoBase[1][0];
+                    gPlayer.speedAirY = 0;
+                } else {
+                    gPlayer.speedAirX = -sKeyboardAccelMusicPlant[1][0];
+                    gPlayer.speedAirY = 0;
+                }
+
+                r7 = Q_8_8(7. / 8.);
+                r5 = -Q_8_8(4.5);
+                r8 = -Q_8_8(7. / 8.);
+                r6 = r5;
+            }
+        } break;
+
+        case 2: {
+            if (kb->unkE > 0) {
+                if (LEVEL_TO_ZONE(gCurrentLevel) == ZONE_6) {
+                    gPlayer.speedAirX = +sKeyboardAccelTechnoBase[2][0];
+                    gPlayer.speedAirY = +sKeyboardAccelTechnoBase[2][1];
+                } else {
+                    gPlayer.speedAirX = +sKeyboardAccelMusicPlant[2][0];
+                    gPlayer.speedAirY = +sKeyboardAccelMusicPlant[2][1];
+                }
+
+                r7 = Q_8_8(7. / 8.);
+                r5 = 0;
+                r8 = -Q_8_8(7. / 8.);
+                r6 = 0;
+            } else if (kb->unkE < 0) {
+                if (LEVEL_TO_ZONE(gCurrentLevel) == ZONE_6) {
+                    gPlayer.speedAirX = +sKeyboardAccelTechnoBase[2][0];
+                    gPlayer.speedAirY = -sKeyboardAccelTechnoBase[2][1];
+                } else {
+                    gPlayer.speedAirX = +sKeyboardAccelMusicPlant[2][0];
+                    gPlayer.speedAirY = -sKeyboardAccelMusicPlant[2][1];
+                }
+
+                r7 = Q_8_8(7. / 8.);
+                r5 = -Q_8_8(4.5);
+                r8 = -Q_8_8(7. / 8.);
+                r6 = r5;
+            } else {
+                if (LEVEL_TO_ZONE(gCurrentLevel) == ZONE_6) {
+                    gPlayer.speedAirX = +sKeyboardAccelTechnoBase[2][0];
+                    gPlayer.speedAirY = 0;
+                } else {
+                    gPlayer.speedAirX = +sKeyboardAccelMusicPlant[2][0];
+                    gPlayer.speedAirY = 0;
+                }
+
+                r7 = Q_8_8(7. / 8.);
+                r5 = -Q_8_8(4.5);
+                r8 = -Q_8_8(7. / 8.);
+                r6 = r5;
+            }
+        } break;
+    }
+
+    if (LEVEL_TO_ZONE(gCurrentLevel) != ZONE_6) {
+        sub_8080C78(Q_24_8_TO_INT(gPlayer.x), Q_24_8_TO_INT(gPlayer.y), 5, 30, (s16)r7,
```

Comment:
Are the casts needed? This is already `s16`

Reply:
Good point! They were initially needed, but I changed the function signature so it might work without, I'll check!